numpy/f2py/tests/test_quoted_character.py

```diff
@@ -28,3 +32,82 @@ class TestQuotedCharacter(util.F2PyTest):
                         reason='Fails with MinGW64 Gfortran (Issue #9673)')
     def test_quoted_character(self):
         assert_equal(self.module.foo(), (b"'", b'"', b';', b'!', b'(', b')'))
+
+@pytest.mark.parametrize("extra_args", [
+    # extra_args can be a list as of gh-11937
+    (['--noopt', '--debug']),
+    # test for string as well, using the same
+    # fcompiler options
+    ('--noopt --debug'),
+    ])
+def test_f2py_init_compile(extra_args):
+    # flush through the f2py __init__
+    # compile() function code path
+    # as a crude test for input handling
+    # following migration from exec_command()
+    # to subprocess.check_output() in gh-11937
+
+    fsource =  b'''
+             integer function foo()
+             foo = 10 + 5
+             return
+             end
+	     '''
+    # use various helper functions in util.py to
+    # enable robust build / compile and
+    # reimport cycle in test suite
+    d = util.get_module_dir()
+    modulename = util.get_temp_module_name()
+
+    # mimic the path changing behavior used
+    # by build_module() in util.py, but don't
+    # actually use build_module() because it
+    # has its own invocation of subprocess
+    # that circumvents the f2py.compile code
+    # block under test
+    cwd = os.getcwd()
+
+    try:
+        os.chdir(d)
+        ret_val = numpy.f2py.compile(fsource,
+                                     modulename=modulename,
+                                     extra_args=extra_args)
+    finally:
+        os.chdir(cwd)
+
+    # check for compile success return value
+    assert_equal(ret_val, 0)
+    # check for sensible result of Fortran function
+    # that means we can import the module name in Python
+    # and retrieve the result of the sum operation
+    return_check = import_module(modulename)
+    calc_result = return_check.foo()
+    assert_equal(calc_result, 15)
+
+def test_f2py_init_compile_failure():
+    # verify an appropriate integer status
+    # value returned by f2py.compile() when
+    # invalid Fortran is provided
+    ret_val = numpy.f2py.compile(b"invalid")
+    assert_equal(ret_val, 1)
+
+def test_f2py_init_compile_bad_cmd():
+    # verify that usage of invalid command in
+    # f2py.compile() returns status value of 127
+    # for historic consistency with exec_command()
+    # error handling
+
+    # patch the sys Python exe path temporarily to
+    # induce an OSError downstream
+    # NOTE: how bad of an idea is this patching?
```

Comment:
I think pytest / standard lib have other ways to monkeypatch as well -- reviewer should perhaps weigh in on this.

Reply:
Is it sufficiently pointless that I can remove catching the `OSError` in the source code itself? If not, it seems like it is sensible to at least probe a code path that raises such an error. Perhaps there's a less aggressive way to raise the error though.